### PR TITLE
2490 fix provider statuses

### LINKED ISSUE #2490: Provider status changes from approved to draft automatically
**Describe the bug**
Tested on v3.29.0-beta.0

**To Reproduce**
1. Create provider in PC, approve provider (use For EPOT privileges)
2. Go to MP - provider is visible _/providers_
3. Refresh the page _/providers_
4. See error - provider disappeared, provider change status to draft (provider is visible in BO)

**Expected behavior**
Provider status isn't changed automatically

### PR DESCRIPTION
**Pull request by kmarszalek**
notyfications: create / update / delete
we use update and delete -> deleted
in update notifications there are: 
1. active approved -> published
2. non-active approved -> draft

Closes #2490

### CODE REVIEW

File: spec/factories/jms_xml_provider.rb [hunk 1]
```diff
@@ -108,4 +108,224 @@
       }
     end
   end
+  factory :jms_xml_draft_provider, class: String do
+    skip_create
```
Comment (jswk) [inline spec/factories/jms_xml_provider.rb:112]: These `skip_create`s won't be needed if you don't call `create(...)` in spec, but `build(...)`.

File: spec/services/jms/manage_message_spec.rb [hunk 2]
```diff
@@ -34,10 +36,39 @@
     $stdout = original_stdout
   end
 
-  it "should receive provider message" do
+  it "should receive update active provider message" do
     original_stdout = $stdout
     $stdout = StringIO.new
     resource = parser.parse(provider_resource["resource"])
+    json_provider =
+      double(body: provider_resource.to_json, headers: { "destination" => "/topic/registry.provider.update" })
```
Comment (jswk) [inline spec/services/jms/manage_message_spec.rb:44]: This should be moved to `let(:json_provider)` above. In the current let, there seems to be a typo in destination though.

File: spec/services/jms/manage_message_spec.rb [hunk 3]
```diff
@@ -34,10 +36,39 @@
...
+    expect { described_class.new(json_provider, eosc_registry_base, logger, nil).call }.to_not raise_error
+    $stdout = original_stdout
+  end
+
+  it "should receive update to draft provider message" do
+    original_stdout = $stdout
+    $stdout = StringIO.new
+    resource = parser.parse(provider_resource["resource"])
+    json_provider =
+      double(body: provider_resource.to_json, headers: { "destination" => "/topic/registry.provider.update" })
```
Comment (jswk) [inline spec/services/jms/manage_message_spec.rb:60]: I would put it in a `let` statement either, as other doubles were.

File: spec/services/jms/manage_message_spec.rb [hunk 4]
```diff
@@ -34,10 +36,39 @@
...
+      resource["providerBundle"]["provider"],
+      resource["providerBundle"]["active"],
+      Time.at(resource["providerBundle"]["metadata"]["modifiedAt"].to_i&./ 1000)
+    )
+
+    expect { described_class.new(json_provider, eosc_registry_base, logger, nil).call }.to_not raise_error
+
+    resource = parser.parse(draft_provider_resource["resource"])
+    json_provider =
+      double(body: draft_provider_resource.to_json, headers: { "destination" => "/topic/registry.provider.update" })
```
Comment (jswk) [inline spec/services/jms/manage_message_spec.rb:71]: I would put it in a `let` statement either, as other doubles were.

File: spec/services/jms/manage_message_spec.rb [hunk 5]
```diff
@@ -48,6 +79,22 @@
     $stdout = original_stdout
   end
 
+  it "should do nothing for update rejected provider message" do
+    original_stdout = $stdout
+    $stdout = StringIO.new
+    resource = parser.parse(rejected_provider_resource["resource"])
+    json_provider =
+      double(body: rejected_provider_resource.to_json, headers: { "destination" => "/topic/registry.provider.update" })
```
Comment (jswk) [inline spec/services/jms/manage_message_spec.rb:87]: I would put it in a `let` statement either, as other doubles were.

**Review by jswk — APPROVED** [review]
Code&logic-wise looks good! I haven't verified it on the actual PC queue though.

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**